Add Why TVC? page to Verifiable Cloud docs (#770)
tkhq/docs: features/verifiable-cloud/why-tvc.mdx

---
title: "Why TVC?"
---

## TEEs don't matter if they aren't run correctly

Anyone can rent a TEE. Almost no one runs them correctly.

> The security of a trusted execution environment is only as strong as everything around it: how code gets approved, how secrets get provisioned, how builds get reproduced, and how upgrades happen. TVC is the only enclave platform where all of it rolls up to the hardware root of trust.

Reviewer: ```suggestion
The security of a trusted execution environment is only as strong as everything around it: how code gets approved, how secrets get provisioned, how builds get reproduced, and how upgrades happen. TVC is the only enclave platform where the process itself rolls up to the hardware root of trust.
```
to specify `it` perhaps

Author: Addressed in 3df86593

## Trust minimization, not trust relocation

> Most TEE platforms remove trust in the cloud provider and hand it to whoever operates the enclave. TVC removes the operator too.

Reviewer: ```suggestion
Most TEE platforms remove trust in the cloud provider and hand it to whoever operates the enclave. TVC removes trust in the operator too.
```

Author: Addressed in 3df86593

* **Quorum approved code upgrades.** No single admin, insider, or compromised account can ship code to an enclave. Every change requires m-of-n cryptographic approval.

> * **Quorum based provisioning for core secrets.** Core secrets only reconstruct inside enclaves running quorum approved code. Each share holder independently verifies the attestation before posting their share.

Author: Addressed in 3df86593

* **Fully reproducible everything.** The operating system ([QuorumOS](https://github.com/tkhq/qos)), the application, and the build system itself ([StageX](https://stagex.tools)) are deterministically reproducible. Anyone can rebuild from source and confirm the running enclave matches, bit for bit.

> * **Verify the whole machine, not just the binary.** Attestation covers the OS and application plus every aspect of configuration: environment variables, CLI arguments, firewall rules, and the set of people who approved the code. The hardware root of trust (the Nitro Secure Module) measures all of it.

Reviewer: ```suggestion
* **Verify the whole machine, not just the binary.** Attestation covers the OS and application plus every aspect of configuration: environment variables, CLI arguments, firewall rules, and the set of people who approved the code. The hardware root of trust (the Nitro Secure Module) accounts for all of it.
```
all of these are nits btw; this one I'm just thinking about what is actually being measured vs included in a chain of trust

Author: Addressed in 3df86593

## A stable identity that survives upgrades

The [Quorum Key](https://github.com/tkhq/qos#quorum-key) gives your application a core secret that persists across software upgrades with zero downtime. Upgrades stay gated by quorum approval, but enclave to enclave attestation carries the key forward automatically, so fleets scale with no human in the loop.

> * **Provably have no access to encrypted user data.** The guarantee comes from cryptography, not from policy.

Reviewer: ```suggestion
* **Operators provably have no access to encrypted user data.** The guarantee comes from cryptography, not from policy.
```
(if you're talking about operators here) i wondered "who" as i read

Author: Addressed in 3df86593

* **Prove that no one else does either.** Your customers, auditors, and regulators can check the proof themselves.

## Fully verified TLS

On TVC, all of the code that verifies TLS rolls up into the Nitro Secure Module measurements, so even your connections to the outside world are part of the proof.

## Secure by default, open by choice

Most confidential VM offerings start as a general purpose machine you must lock down. TVC starts from the most locked down possible configuration, built for processing targets holding billions of dollars. You selectively enable functionality like egress as you need it. Subtractive security fails quietly. Additive security fails safe.

## Part of the Turnkey stack

TVC interoperates natively with Turnkey's secret and key management system: policy gated key release, arbitrary consensus thresholds, and policies that can bind key access to exact application hashes and attestation contents.

## Get started

> Make a Turnkey account, download the [TVC CLI](https://github.com/tkhq/rust-sdk/tree/main/tvc), and get everything above running with one prompt. See the [quickstart](/features/verifiable-cloud/quickstart) to deploy your first app.

Reviewer: ```suggestion
Make a Turnkey account and get everything above running with one prompt. See the [quickstart](/features/verifiable-cloud/quickstart) to deploy your first app.
```
since the quickstart includes CLI installation in one command (vs directing to repo with more confusing instructions)

Author: Addressed in 3df86593

> TVC is currently in private beta. [Join the waitlist](https://www.turnkey.com/turnkey-verifiable-cloud#waitlist) to request access.

Reviewer 2: Let's reuse the callout component for TVC private beta here: `/snippets/tvc-beta-callout.mdx`

```suggestion
import { TvcBetaCallout } from "/snippets/tvc-beta-callout.mdx";

<TvcBetaCallout />
```

Author: Addressed in 3df86593

> TVC is currently in private beta. [Join the waitlist](https://www.turnkey.com/turnkey-verifiable-cloud#waitlist) to request access.

Author: Addressed in 3df86593
